my-fund-backend/src/main/java/com/myfund/model/DTO/mapper/CategoryMapper.java

```diff
@@ -0,0 +1,73 @@
+package com.myfund.model.DTO.mapper;
+
+import com.myfund.model.Category;
+import com.myfund.model.DTO.CreateSubCategoryDTO;
+import com.myfund.model.DTO.CategoryDTO;
+import com.myfund.model.DTO.CreateCategoryDTO;
+import com.myfund.model.DTO.SubCategoryDTO;
+import com.myfund.model.SubCategory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CategoryMapper {
+    public static Category createCategoryDTOMapToCategory(CreateCategoryDTO createCategoryDTO) {
+        Category category = new Category();
+        ArrayList<SubCategory> subCategoryList = new ArrayList<>();
+        if (createCategoryDTO.getSubCategories() != null) {
+            for (CreateSubCategoryDTO createSubCategoryDTO : createCategoryDTO.getSubCategories()) {
+                SubCategory subCategory = createSubCategoryDTOToSubCategory(createSubCategoryDTO);
+                subCategoryList.add(subCategory);
+            }
+        }
+        category.setName(createCategoryDTO.getName());
+        category.setSubCategories(subCategoryList);
+        return category;
+    }
+
+    private static SubCategory createSubCategoryDTOToSubCategory(CreateSubCategoryDTO createSubCategoryDTO) {
+        SubCategory subCategory = new SubCategory();
+        subCategory.setName(createSubCategoryDTO.getName());
+        return subCategory;
+    }
+
+    public static CategoryDTO categoryMapToCategoryDTO(Category category) {
```

Comment:
hmm, maybe it should be a part of Category class? 
then it will be easier to call category.mapToDto(...) ? not sure about it, please read more about it

Reply:
ok, I think now it is better to have it in mapper as you have

Reply:
ok